CodeEditModules/Modules/Keybindings/src/KeybindingManager.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  KeybindingManager.swift
+//  
+//
+//  Created by Alex on 09.05.2022.
+//
+
+import Foundation
+import SwiftUI
+
+// swiftlint:disable missing_docs
+public final class KeybindingManager {
+    public var keyboardShortcuts = [String: KeyboardShortcutWrapper]()
+
+    private init() {
+        loadKeybindings()
+    }
+
+    // swiftlint:disable missing_docs
+    public static let shared: KeybindingManager = .init()
+
+    // We need this fallback shortcut because optional shortcuts available only from 12.3, while we have target of 12.0x
+    var fallbackShortcut = KeyboardShortcutWrapper(name: "?", description: "Test", context: "Fallback",
+                                                   keybinding: "?", modifier: "shift", id: "fallback")
+
+    public func addNewShortcut(shortcut: KeyboardShortcutWrapper, name: String) {
+        keyboardShortcuts[name] = shortcut
+    }
+
+    private func loadKeybindings() {
+
+        let bindingsURL = Bundle.module.url(forResource: "default_keybindings.json", withExtension: nil)
+        if let json = try? Data(contentsOf: bindingsURL!) {
+            do {
+                let prefs = try JSONDecoder().decode([KeyboardShortcutWrapper].self, from: json)
+                for pref in prefs {
+                    addNewShortcut(shortcut: pref, name: pref.id)
+                }
+                } catch {
+                    print("error:\(error)")
+                }
+        }
+            return
+//        let preferenceURL = AppPreferencesModel.shared.preferencesURL
+    }
+
+    private func loadDefaultKeybindings() {
+
+    }
+
+    /// Get shortcut by name
+    /// - Parameter name: shortcut name
+    /// - Returns: KeyboardShortcutWrapper
+    public func named(with name: String) -> KeyboardShortcutWrapper {
+        let foundElement = keyboardShortcuts[name]
+        return foundElement != nil ? foundElement! : fallbackShortcut
+    }
+
+}
+
+public struct KeyboardShortcutWrapper: Codable {
```

Comment:
Please add proper documentation to `public` classes, structs, methods and variables/constants.

A guide about code documentation can be found [here](https://nshipster.com/swift-documentation/).